gpu/containers/src/initialization.cpp

```diff
@@ -115,16 +116,13 @@ namespace
             int Cores;
         };
 
-        SMtoCores gpuArchCoresPerSM[] = {
+        std::array<SMtoCores, 13> gpuArchCoresPerSM = {{
             {0x10,   8}, {0x11,   8}, {0x12,   8}, {0x13,  8}, {0x20,  32}, {0x21, 48}, {0x30, 192},
-            {0x35, 192}, {0x50, 128}, {0x52, 128}, {0x53, 128}, {0x60, 64}, {0x61, 128}, {-1, -1}
-        };
-        int index = 0;
-        while (gpuArchCoresPerSM[index].SM != -1) 
-        {
-            if (gpuArchCoresPerSM[index].SM == ((major << 4) + minor) ) 
-                return gpuArchCoresPerSM[index].Cores;
-            index++;
+            {0x35, 192}, {0x50, 128}, {0x52, 128}, {0x53, 128}, {0x60, 64}, {0x61, 128}
```

Comment:
We should add more here. 62, 70, 72, 75, 80, 86 are the new architectures (Pascal, Volta, Turing, Ampere)

Reply:
Seems a lot like this: https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L652-L668.

Reply:
Thank you very much, @kunaltyagi  and @larshg, for your comments! Your comments were of great help to me! I added the new architectures. I am still wondering about two aspects:
- Don't we support two Kepler architectures? We are missing values of 32 and 37 in comparison to the link Lars provided. Plus, we mention older architectures (until 30) in the list. Do we know if our code compiles with these architectures? Should we maybe remove these values? 
- Why do our original code and the NVIDIA code check for {-1,-1} at the end. Isn't that redundant?

Reply:
* Based on this (https://en.wikipedia.org/wiki/CUDA#GPUs_supported), we could add the older versions, but we'd have to remove them again when CUDA 11 becomes the oldest supported version. We can add 37 (but not before 35)
* That's not needed anymore thanks to your refactoring

Reply:
Our oldest CUDA support is for CUDA 9 I believe. @larshg can we remove pre 3.0?

Reply:
We don't compile for less than 3.0 as of our CMake https://github.com/PointCloudLibrary/pcl/blob/master/cmake/pcl_find_cuda.cmake#L33-L39.

So I don't see any reason not to remove pre 3.0.

Reply:
Thank you again, @kunaltyagi  and @larshg, for your feedback! I harmonized the information in `pcl_find_cuda` and `initialization.cpp`. All compute capability versions starting from 3.0 are now included. Do you agree with the changes?